File: src/TypeFormatter/NumberTypeFormatter.ts
import type { Definition } from "../Schema/Definition.js";
import type { SubTypeFormatter } from "../SubTypeFormatter.js";
import type { BaseType } from "../Type/BaseType.js";
import { NumberType } from "../Type/NumberType.js";

export class NumberTypeFormatter implements SubTypeFormatter {
    public supportsType(type: BaseType): boolean {
        return type instanceof NumberType;
    }
    public getDefinition(type: NumberType): Definition {
        return { type: "number" };
    }
    public getChildren(type: NumberType): BaseType[] {
        return [];
    }
}

File: internal/formatter/number_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/schema"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// NumberTypeFormatter mirrors src/TypeFormatter/NumberTypeFormatter.ts.
type NumberTypeFormatter struct{}

func NewNumberTypeFormatter() *NumberTypeFormatter { return &NumberTypeFormatter{} }

func (f *NumberTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.NumberType)
	return ok
}

func (f *NumberTypeFormatter) GetDefinition(t types.Type) *schema.Definition {
	return &schema.Definition{Type: "number"}
}

func (f *NumberTypeFormatter) GetChildren(t types.Type) []types.Type { return nil }
